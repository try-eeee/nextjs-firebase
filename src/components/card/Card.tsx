import styles from "./Card.module.scss";

interface CardProps {
  title: string;
  children: React.ReactNode;
}

export function Card(props: CardProps) {
  const { title, children } = props;
  return (
    <div {...props} className={styles.wrapper}>
      <div className={styles.title}>{title}</div>
      {children}
    </div>
  );
}
