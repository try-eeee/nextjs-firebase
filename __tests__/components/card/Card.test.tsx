import React from "react";
import { render } from "@testing-library/react";
import { Card } from "@/components/card/Card";

describe("Card UnitTest", () => {
  test("Snapshot Test Of Card", () => {
    const { asFragment } = render(
      <Card title="新規登録">
        <>コンテンツ</>
      </Card>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
