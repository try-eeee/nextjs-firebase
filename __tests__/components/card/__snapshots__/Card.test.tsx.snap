// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card UnitTest Snapshot Test Of Card 1`] = `
<DocumentFragment>
  <div
    class="wrapper"
    title="新規登録"
  >
    <div
      class="title"
    >
      新規登録
    </div>
    コンテンツ
  </div>
</DocumentFragment>
`;
